src/pybind/mgr/cephadm/http_server.py

```diff
@@ -37,17 +39,47 @@ def configure_cherrypy(self) -> None:
             'engine.autoreload.on': False,
         })
 
-    def run(self) -> None:
+    def configure(self) -> None:
+        self.configure_cherrypy()
+        self.agent.configure()
+        self.service_discovery.configure(self.mgr.service_discovery_port, self.mgr.get_mgr_ip())
+
+    def config_update(self) -> None:
+        self.service_discovery_port = self.mgr.service_discovery_port
```

Comment:
yeah, since this is the ctr I'm using the attribute directly (instead of the property) to avoid calling the underlying code. Outside the ctr the property is used instead.